CartAPI/Repository/CartRepository.cs

```diff
@@ -27,55 +30,24 @@ public async Task<CartResult> CreateCart()
             _cartContext.Carts.Add(cart);
             await _cartContext.SaveChangesAsync();
 
-            var cartResult = new CartResult()
-            {
-                Id = cart.Id,
-                State= cart.State.ToString()
-            };
+            var cartResult = _mapper.Map<CartResult>(cart);
             return cartResult;
         }
 
         public async Task<List<CartResult>> GetAllCartsAsync()
         {
-            var results = await _cartContext.Carts.Include(o=>o.Items).Select(c => new CartResult()
-            {
-                Id = c.Id,
-                State = c.State.ToString(),
-                Items = c.Items.Select(ci=>new ItemResult()
-                {
-                    Id = ci.Id,
-                    Title = ci.Title,
-                    Price = ci.Price,
-                    Quantity = ci.Quantity,
-                    CartId = ci.CartId,
-                    ExternalId = ci.ExternalId
-                }).ToList()
-            }).ToListAsync();
+            var results = _mapper.Map<List<CartResult>>(await _cartContext.Carts.Include(o => o.Items).ToListAsync());
             return results;
         }
 
-        public async Task<CartResult> GetCartById(long id)
+        public async Task<CartResult> GetCartById(Guid id)
         {
-            var result = await _cartContext.Carts.Include(o=>o.Items).FirstOrDefaultAsync(c => c.Id == id);
+            var result = _mapper.Map<CartResult>(await _cartContext.Carts.Include(o => o.Items).FirstOrDefaultAsync(c => c.Id == id));
```

Comment:
Separate value, + null check should be on the DB result not on the mapped result.